avatar/metrics/trace.py

```diff
@@ -0,0 +1,241 @@
+# Copyright 2022 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Avatar metrics trace."""
+
+import re
+import time
+import types
+
+from avatar.metrics.trace_pb2 import (
+    DebugAnnotation,
+    ProcessDescriptor,
+    ThreadDescriptor,
+    Trace,
+    TracePacket,
+    TrackDescriptor,
+    TrackEvent,
+)
+from google.protobuf.text_encoding import CUnescape
+from mobly.base_test import BaseTestClass
+from typing import TYPE_CHECKING, Any, List, Protocol, Union
+
+if TYPE_CHECKING:
+    from avatar import PandoraDevices
+    from avatar.pandora_client import PandoraClient
+else:
+    PandoraClient = object
+    PandoraDevices = object
+
+
+packets: List[TracePacket] = []
+genesis: int = time.monotonic_ns()
+uuid: int = 0
+
+
+def process_name(test: BaseTestClass) -> str:
+    return f"{test.__class__.__name__}.{test.current_test_info.name}"
+
+
+def next_uuid() -> int:
```

Comment:
nit: `next_id`, a `uuid` is a different thing

Reply:
Well yeah I agree, just because they were unique in some point, I'll change it